File: Clases/Unidad02_POO/C#/Publicaciones_v03/Publicaciones_v03/Articulo.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Publicaciones_v03
{
    internal class Articulo : Publicacion
    {
        // Atributos
        private string revista;
        private int volumen;
        private int numero;
        private string mes;
        // Metodos
        // Constructor
        public Articulo(string nombre, string[] autores, int cantidad_autores, int año,
            string revista, int volumen, int numero, string mes) :
            base(nombre, autores, cantidad_autores, año)
        {
            this.revista = revista;
            this.volumen = volumen;
            this.numero = numero;
            this.mes = mes;
        }
        public Articulo(string nombre, string autor, int año,
            string revista, int volumen, int numero, string mes) :
            base(nombre, autor, año)
        {
            this.revista = revista;
            this.volumen = volumen;
            this.numero = numero;
            this.mes = mes;
        }
        // Impresion
        public override void referenciar()
        {
            string referencia = "[" + Publicacion.orden_presentacion + "] ";
            for (int i = 0; i < this.cantidad_autores; i++)
            {
                if (i != 0)
                {
                    if (i == this.cantidad_autores - 1) referencia += " and ";
                    else referencia += ", ";
                }
                referencia += this.autores[i];
            }
            referencia += ", \"" + this.nombre + "\", " + this.revista +
                ", vol. " + this.volumen.ToString() + ", no. " + this.numero.ToString() +
                ", " + this.mes + ", " + this.año.ToString() + ". ";
            Console.WriteLine(referencia);
        }
    }
}

File: Clases/Unidad03_Librerias_Componentes/Java/CompilacionManual/PaquetesEnLibrerias/src/Publicaciones_v03_b/publicaciones/estandardAPA/Articulo.java
package Publicaciones_v03_b.publicaciones.estandardAPA;

import Publicaciones_v03_b.publicaciones.Publicacion;

public class Articulo extends Publicacion{
// Atributos
    private String revista;
    private Integer volumen;
    private Integer numero;
    private String mes;
// Metodos
    // Constructor
    public Articulo(String nombre, String autores[], Integer año, String revista, Integer volumen, Integer numero, String mes){
        super(nombre, autores, año);
        this.revista = revista;
        this.volumen = volumen;
        this.numero = numero;
        this.mes = mes;
    }
    public Articulo(String nombre, String autor, Integer año, String revista, Integer volumen, Integer numero, String mes){
        super(nombre, autor, año);
        this.revista = revista;
        this.volumen = volumen;
        this.numero = numero;
        this.mes = mes;
    }
    // Impresion
    @Override
    public void referenciar(){
        String referencia = "";
        for(int i=0; i<this.cantidad_autores; i++){
            if (i!=0){
                if (i==this.cantidad_autores-1){referencia += " and ";}
                else{referencia += ", ";}
            }
            referencia += this.autores.get(i).toString();                    
        }
        referencia += ", (" + año.toString() + "). " + this.nombre + ". " + this.revista;
        referencia += ", " + volumen.toString() + "(" + this.numero.toString() + ").";
        System.out.println(referencia);
    }
}
